core/controllers/story_editor.py

```diff
@@ -126,10 +141,41 @@ def delete(self, topic_id, story_id):
         story_domain.Story.require_valid_story_id(story_id)
         topic_domain.Topic.require_valid_topic_id(topic_id)
 
-        story = story_services.get_story_by_id(story_id, strict=False)
+        story = story_fetchers.get_story_by_id(story_id, strict=False)
         if story is None:
             raise self.PageNotFoundException
 
         story_services.delete_story(self.user_id, story_id)
 
         self.render_json(self.values)
+
+
+class StoryPublishHandler(base.BaseHandler):
+    """A data handler for publishing and unpublishing stories."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_story
+    def put(self, topic_id, story_id):
+        """Published/unpublished given story."""
+        story_domain.Story.require_valid_story_id(story_id)
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+        story = story_fetchers.get_story_by_id(story_id, strict=False)
+        if story is None:
+            raise self.PageNotFoundException
+
+        topic = topic_services.get_topic_by_id(topic_id, strict=False)
+        canonical_story_ids = topic.get_canonical_story_ids()
+        if topic is None or story_id not in canonical_story_ids:
+            raise self.PageNotFoundException
+
+        story_publication_status = self.payload.get('story_publication_status')
```

Comment:
Consider calling this new_story_status_is_public or similar -- see other comment.

Reply:
Done